astropy/io/fits/hdu/compressed/_tiled_compression.py

```diff
@@ -465,7 +467,11 @@ def decompress_image_data_section(
             if zblank is not None:
                 if not tile_data.flags.writeable:
                     tile_data = tile_data.copy()
-                tile_data[blank_mask] = np.nan
+                if zbitpix > 0:
+                    blank = zblank_header
+                else:
+                    blank = np.nan
```

Comment:
```suggestion
                blank = zblank_header if zbitpix > 0 else np.nan
```

Reply:
Given the shortening, this might also be combined with the following line

Reply:
OK, made that change too.  I'm not crazy about the 1-line style for assignments but it is fairly readable in this case.